rust/src/ffi/base64.rs

```diff
@@ -127,6 +127,28 @@ pub unsafe extern "C" fn SCBase64Decode(
     return num_decoded;
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn SCBase64DecodeLenient(
+    input: *const u8, len: usize, output: *mut u8,
+) -> u32 {
+    if input.is_null() || len == 0 {
+        return 0;
+    }
+
+    let in_vec = build_slice!(input, len);
+    let out_vec = std::slice::from_raw_parts_mut(output, len);
+    let mut num_decoded: u32 = 0;
+    let config = base64::engine::GeneralPurposeConfig::new()
+        .with_decode_padding_mode(base64::engine::DecodePaddingMode::Indifferent);
+    let decoder = base64::engine::GeneralPurpose::new(&base64::alphabet::STANDARD, config);
```

Comment:
How lenient is this ? Like tolerating spaces inside the base64 ?

Reply:
Just with respect to padding, say you don't know if your chunk has padding or not.  Given STANDARD and STANDARD_NO_PAD are strict about the padding.  Perhaps a better name?

Reply:
I've renamed this to be PadOpt.. Lenient should be saved for a decoder that does a best effort to extract whatever it can.

Reply:
Thanks